docs/framework/wcf/samples/custom-lifetime.md

```diff
@@ -1,219 +1,227 @@
-  
+> The setup procedure and build instructions for this sample are located at the end of this article.
+
+## Shared instancing
+
+WCF offers several instancing modes for your service instances. The shared instancing mode covered in this article provides a way to share a service instance between multiple channels. Clients can contact a factory method in the service and create a new channel to start communication. The following code snippet shows how a client application creates a new channel to an existing service instance:
+
+```csharp
+// Create the first channel.
+IEchoService proxy = channelFactory.CreateChannel();
+
+//...
+
+// Close the first channel
+((IChannel)proxy).Close();
+
+// Send another request from a different channel using the same instance.
+proxy = channelFactory.CreateChannel();
+```
+
+Unlike other instancing modes, the shared instancing mode has a unique way of releasing the service instances. When all the channels are closed for an instance, the service WCF runtime starts a timer. If nobody makes a connection before the timeout expires, WCF releases the instance and claims the resources. As a part of the teardown procedure WCF invokes the <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider.IsIdle%2A> method of all <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider> implementations before releasing the instance. If all of them return `true` the instance is released. Otherwise the <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider> implementation is responsible for notifying the `Dispatcher` of the idle state by using a callback method.
+
+By default, the idle timeout value of <xref:System.ServiceModel.InstanceContext> is one minute. However, this sample demonstrates how you can extend this using a custom extension.
+
+## Extending the InstanceContext
+
+In WCF, <xref:System.ServiceModel.InstanceContext> is the link between the service instance and the `Dispatcher`. WCF allows you to extend this runtime component by adding new state or behavior by using its extensible object pattern. The extensible object pattern is used in WCF to either extend existing runtime classes with new functionality or to add new state features to an object. There are three interfaces in the extensible object pattern: <xref:System.ServiceModel.IExtensibleObject%601>, <xref:System.ServiceModel.IExtension%601>, and <xref:System.ServiceModel.IExtensionCollection%601>.
+
+The <xref:System.ServiceModel.IExtensibleObject%601> interface is implemented by objects to allow extensions that customize their functionality.
+
+The <xref:System.ServiceModel.IExtension%601> interface is implemented by objects that can be extensions of classes of type `T`.
+
+And finally, the <xref:System.ServiceModel.IExtensionCollection%601> interface is a collection of [IExtensions](xref:System.ServiceModel.IExtension%601) that allows for retrieving [IExtensions](xref:System.ServiceModel.IExtension%601) by their type.
```

Comment:
It should be IExtension<T> and not IExtensions. Grammar can get complicated when dealing with abstract concepts like type hierarchies but I've done my best for suggesting who it can be rewritten. Feel free to improve upon this further, the key point is that the type should be correctly listed.

is a collection of [IExtension<T>](xref:System.ServiceModel.IExtension%601) implementations that allows for retrieving an implementation of [IExtension<T>](xref:System.ServiceModel.IExtension%601) by their type.